Source/dx.cpp

```diff
@@ -133,22 +164,27 @@ void __cdecl dx_create_primary_surface()
 	ddsd.dwSize = sizeof(ddsd);
 	ddsd.dwFlags = DDSD_CAPS;
 	ddsd.ddsCaps.dwCaps = DDSCAPS_PRIMARYSURFACE;
+#ifdef __cplusplus
 	error_code = lpDDInterface->CreateSurface(&ddsd, &lpDDSPrimary, NULL);
+#else
+	error_code = lpDDInterface->lpVtbl->CreateSurface(lpDDInterface, &ddsd, &lpDDSPrimary, NULL);
+#endif
 	if (error_code != DD_OK)
 		ErrDlg(IDD_DIALOG1, error_code, "C:\\Src\\Diablo\\Source\\dx.cpp", 109);
 }
 
 HRESULT __fastcall dx_DirectDrawCreate(LPGUID guid, LPDIRECTDRAW *lplpDD, LPUNKNOWN pUnkOuter)
 {
+	HRESULT(WINAPI * DirectDrawCreate)

```

Comment:
Ahhh, that makes sense. The function pointer just had to be declared at the top.

Reply:
Yeah a bit after you mentioned that "it's just a pointer" it suddenly made sens :)